inst/schema/spec.md

```diff
@@ -0,0 +1,229 @@
+
+# Data
+## POST /data/tab-name/
+Accepts a set of baseline options, and returns a data set for the data under the named tab.
+
+Request schema: [DataOptions.schema.json](./DataOptions.schema.json)
+
+Return schema: [Data.schema.json](./Data.schema.json)
+
+### Example
+#### Request
+```
+{
+  "irs_future": "80%",
+  "sprayInput": 1,
+  "biting_indoors": "high",
+  "population": 1000
+}
+```
+
+#### Response
+```
+[
+    {
+        "intervention": "none",
+        "net_use": 20,
+        "prev_year_1": 7.32,
+        "prev_year_2": 7.34,
+        "prev_year_3": 7.48,
+        "cases_averted": 0,
+        "cases_averted_high": 0,
+        "cases_averted_low": 0,
+    },
+    {
+        "intervention": "none",
+        "net_use": 40,
+        "prev_year_1": 7.32,
+        "prev_year_2": 7.34,
+        "prev_year_3": 7.48,
+        "cases_averted": 0,
+        "cases_averted_high": 0,
+        "cases_averted_low": 0,
+    },
+    {
+        "intervention": "ITN",
+        "net_use": 20,
+        "prev_year_1": 5.32,
+        "prev_year_2": 5.34,
+        "prev_year_3": 5.48,
+        "cases_averted": 51,
+        "cases_averted_high": 60,
+        "cases_averted_low": 50,
+    },
+    {
+        "intervention": "ITN",
+        "net_use": 40,
+        "prev_year_1": 4.32,
+        "prev_year_2": 4.34,
+        "prev_year_3": 4.48,
+        "cases_averted": 52,
+        "cases_averted_high": 62,
+        "cases_averted_low": 50,
+    },
+    {
+        "intervention": "PBO",
+        "net_use": 20,
+        "prev_year_1": 6.32,
+        "prev_year_2": 6.34,
+        "prev_year_3": 6.48,
+        "cases_averted": 61,
+        "cases_averted_high": 70,
+        "cases_averted_low": 60,
+    },
+    {
+        "intervention": "PBO",
+        "net_use": 40,
+        "prev_year_1": 5.32,
+        "prev_year_2": 5.34,
+        "prev_year_3": 5.48,
+        "cases_averted": 62,
+        "cases_averted_high": 72,
+        "cases_averted_low": 60,
+    }
+]
+```
+
+# Tables
+## GET /table/:tab-name/
+Returns metadata about the columns for the table to be displayed under this tab.
+
+Schema: [TableDefinition.schema.json](./TableDefinition.schema.json)
+
+```
+[
+    {"id": "intervention", "name": "Interventions"},
+    {"id": "net_use", "name": "Net use"},
+    {"id": "prev_year_1", "name": "Prevalence Under 5 yrs: Yr 1 post intervention"},
+    {"id": "prev_year_2", "name": "Prevalence Under 5 yrs: Yr 2 post intervention"},
+    {"id": "prev_year_3", "name": "Prevalence Under 5 yrs: Yr 3 post intervention"},
+    {"id": "cases_averted", "name": "Cases averted across 3 yrs since intervention"}
+]
+```
+
+# Graphs
+
+## GET /graphs/:tab-name/
+Returns an array of graph metadata objects for the named tab. Each graph metadata object must have:
+* an id
+* a plotly layout object
+* a data object containing all the plotly metadata for the data series as well as the columns defining the series
+
+and optionally:
+* a plotly config object
```

Comment:
Does the plotly config belong here, or do we want to use some higher level config concepts that we map into plotly? Or is this something that the backend even changes between calls and should be done on app startup?

Reply:
This is the part I'm least sure about. But this approach has the advantage that the researchers can effectively change the graphs to anything they want just by changing the R API package.